src/js/components/Beacon/BeaconCommon/BeaconQueryFormUi.tsx

```diff
@@ -67,12 +70,18 @@ const BeaconQueryFormUi = ({
   const showError = hasError && !errorAlertClosed;
 
   const requestPayload = useCallback(
-    (query: PayloadVariantsQuery, payloadFilters: PayloadFilter[]): BeaconQueryPayload => ({
-      meta: { apiVersion: '2.0.0' },
-      query: { requestParameters: { g_variant: query }, filters: payloadFilters },
-      bento: { showSummaryStatistics: true },
-    }),
-    []
+    (query: PayloadVariantsQuery, payloadFilters: PayloadFilter[]): BeaconQueryPayload => {
+      const payload: BeaconQueryPayload = {
+        meta: { apiVersion: '2.0.0' },
+        query: { requestParameters: { g_variant: query }, filters: payloadFilters },
+        bento: { showSummaryStatistics: true },
+      };
+      if (scope.dataset) {
+        payload['query']['datasets'] = { datasetIds: [scope.dataset] };
+      }
+      return payload;
+    },
+    [scope]
   );
 
   const launchEmptyQuery = useCallback(
```

Comment:
can't comment on the below, but the empty query should only launch (with the useeffect below) when the scope has been successfully set from the URL via `scopeSet`. I think this should fix some of the re-rendering.